### ISSUE TITLE
Is it possible to extend the tailwind theme with the twindv1 plugin?

### ISSUE THREAD

**Issue by itanglo** [denoland/fresh#1633]
I tried adding the following to the `twind.config.ts` file:

```ts
import { defineConfig, Preset } from "https://esm.sh/@twind/core@1.1.3";
import presetTailwind from "https://esm.sh/@twind/preset-tailwind@1.1.4";
import presetAutoprefix from "https://esm.sh/@twind/preset-autoprefix@1.0.7";

export default {
  ...defineConfig({
    presets: [presetTailwind() as Preset, presetAutoprefix()],
    theme: {
      extend: {
        colors: {
          primary: '#4fa846'
        }
      }
    }
  }),
  selfURL: import.meta.url,
};
```

Is this the correct way to extend the tailwind theme using the twindv1 plugin?

It doesn't seem to work for me and I get a type error in the `main.ts` file when trying to run it:

```ts
import { start } from "$fresh/server.ts";
import manifest from "./fresh.gen.ts";

import twindPlugin from "$fresh/plugins/twindv1.ts";
import twindConfig from "./twind.config.ts";

await start(manifest, { plugins: [twindPlugin(twindConfig)] });
```

```sh
Argument of type '{ selfURL: string; darkMode?: DarkModeConfig; darkColor?: DarkColor<BaseTheme & { extend: never; }> | undefined; theme: ThemeConfig<BaseTheme & { extend: never; }>; ... 6 more ...; finalize: Finalize<...>[]; }' is not assignable to parameter of type 'Options'.
  Types of property 'darkColor' are incompatible.
    Type 'DarkColor<BaseTheme & { extend: never; }> | undefined' is not assignable to type 'DarkColor<BaseTheme> | undefined'.
      Type 'DarkColor<BaseTheme & { extend: never; }>' is not assignable to type 'DarkColor<BaseTheme>'.
        Types of parameters 'context' and 'context' are incompatible.
          Type 'Context<BaseTheme>' is not assignable to type 'Context<BaseTheme & { extend: never; }>'.
            The types returned by 'theme()' are incompatible between these types.
              Type 'BaseTheme' is not assignable to type 'BaseTheme & { extend: never; }'.deno-ts(2345)
```

**Comment by deer** [comment]
It seems like they have some typing issues all over the place. If you force it like this:
```tsx
    theme: {
      extend: {
        colors: {
          primary: "#4fa846",
        },
      },
    } as ThemeConfig,
```
then the error goes away.

The `theme` property of a TwindUserConfig is typed like this:
```tsx
theme?: Theme | ThemeConfig<BaseTheme & ExtractThemes<Theme, Presets>>;
```
and a `ThemeConfig` is:
```tsx
declare type ThemeConfig<Theme extends BaseTheme = BaseTheme> = PartialTheme<Theme> & {
    extend?: PartialTheme<Theme>;
};
```

Let me know if this helps.

**Comment by itanglo** [comment]
Thanks, @deer 

That suggestion has got rid of the type error.

I actually realised that the change I made to the theme was working in spite of the type error, but I was using the CSS class name incorrectly. When I combined the color I specified with a valid class name then it worked. 

For example:

`<h1 class="text-primary">Hello</h1>`

The problem was I was trying to use it like this:

`<h1 class="primary">Hello</h1>`

Great to have got the error out of the way too though as I would have found that annoying. Thanks for your help.
